python/GafferOSLUI/OSLCodeUI.py

```diff
@@ -0,0 +1,320 @@
+			flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic
+		)
+
+		with Gaffer.UndoContext( self.getPlug().ancestor( Gaffer.ScriptNode ) ) :
+			self.getPlug().addChild( plug )
+
+##########################################################################
+# _CodePlugValueWidget
+##########################################################################
+
+class _CodePlugValueWidget( GafferUI.MultiLineStringPlugValueWidget ) :
+
+	def __init__( self, plug, **kw ) :
+
+		GafferUI.MultiLineStringPlugValueWidget.__init__( self, plug, **kw )
+
+		self.textWidget().setRole( GafferUI.MultiLineTextWidget.Role.Code )
+
+		self.__dropTextConnection = self.textWidget().dropTextSignal().connect( Gaffer.WeakMethod( self.__dropText ) )
+
+	def __dropText( self, widget, dragData ) :
+
+		if not isinstance( dragData, Gaffer.Plug ) :
+			return None
+
+		plug = dragData
+		node = plug.node()
+		if plug.parent() not in ( node["parameters"], node["out"] ) :
+			return None
+
+		if isinstance( plug, Gaffer.SplinefColor3fPlug ) :
+			return "colorSpline( {0}Positions, {0}Values, {0}Basis, u )".format( plug.getName() )
+
+		return plug.getName()
+
+##########################################################################
+# _ErrorWidget
+##########################################################################
+
+class _ErrorWidget( GafferUI.Widget ) :
+
+	def __init__( self, node, **kw ) :
+
+		self.__messageWidget = GafferUI.MessageWidget()
+		GafferUI.Widget.__init__( self, self.__messageWidget, **kw )
+
+		self.__errorConnection = node.errorSignal().connect( Gaffer.WeakMethod( self.__error ) )
+		self.__shaderCompiledConnection = node.shaderCompiledSignal().connect( Gaffer.WeakMethod( self.__shaderCompiled ) )
+
+		self.__messageWidget.setVisible( False )
+
+	def __error( self, plug, source, error ) :
+
+		self.__messageWidget.clear()
+		self.__messageWidget.messageHandler().handle( IECore.Msg.Level.Error, "Compilation error", error )
+		self.__messageWidget.setVisible( True )
+
+	def __shaderCompiled( self ) :
+
+		self.__messageWidget.setVisible( False )
+
+##########################################################################
+# Plug menu
+##########################################################################
+
+## \todo This functionality is duplicated in several places (NodeUI,
+#  BoxUI, CompoundDataPlugValueWidget). It would be better if we could
+#  just control it in one place with a "plugValueWidget:removeable"
+#  metadata value. This main reason we can't do that right now is that
+#  we'd want to register the metadata with "parameters.*", but that would
+#  match "parameters.vector.x" as well as "parameters.vector". This is
+#  a general problem we have with the metadata matching - we should make
+#  '.' unmatchable by '*'.
+def __deletePlug( plug ) :
+
+	with Gaffer.UndoContext( plug.ancestor( Gaffer.ScriptNode ) ) :
+		plug.parent().removeChild( plug )
+
+def __plugPopupMenu( menuDefinition, plugValueWidget ) :
```

Comment:
Thanks. If it does come up super soon, it'll be a quick job to chuck in some menu items. It might not even be that big a job to get the UI Editor working with it...